### PR TITLE
Migrate helm chart from helm v2 to v3

### PR DESCRIPTION
**Pull request by AWaterColorPen**
#### Rename directory from `helm` to `helm/druid`. 
- It should place a folder named `druid` under `helm` folder to do **helm package**, just like https://github.com/apache/superset/tree/master/helm/superset
#### Migrate helm chart from helm v2 to v3
- change `apiVersion` from `v1` to `v2`
- migrate dependencies to Chart.yaml

This PR has:
- [x] been self-reviewed.
   - [x] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] added integration tests.
- [x] been tested in a test Druid cluster.

cc @asdf2014 @maver1ck

### CODE REVIEW

File: helm/druid/Chart.yaml [hunk 1]
```diff
@@ -13,10 +13,23 @@
...
+    condition: zookeeper.enabled
+  - name: mysql
+    version: 1.6.4
+    repository: https://charts.helm.sh/stable
+    condition: mysql.enabled
+  - name: postgresql
+    version: 8.6.4
+    repository: https://charts.helm.sh/stable
+    condition: postgresql.enabled
 version: 0.2.18
```
Comment (maver1ck) [inline helm/druid/Chart.yaml:33]: I think we need to bump version

Comment (AWaterColorPen) [reply helm/druid/Chart.yaml:33]: oh, we need to bump version. BTW, since migrating from v2 to v3, I think we can bump to `0.3.0`

**Review by maver1ck — APPROVED** [review]

**Review by asdf2014 — APPROVED** [review]

**Review by QiuMM — APPROVED** [review]

**Comment by richardlikeswasabi** [comment]
This is awesome, are there plans to maintain an official Helm chart for Druid?

**Comment by jihoonson** [comment]
@asdf2014 Can you please check my last comment in the dev thread?

**Comment by asdf2014** [comment]
Hi, @jihoonson. Sure, let me check it out. Thank you for your reminder.

**Pull request merged**